src/streams/StreamItem.js

```diff
@@ -44,6 +44,7 @@ type Props = $ReadOnly<{|
   isMuted: boolean,
   isPrivate: boolean,
   isSubscribed?: boolean,
+  isWebPublic?: boolean,
```

Comment:
nit: Let's be consistent between this prop on StreamItem, the prop on StreamIcon, and the property in UnreadStreamItem objects, about
* `isWebPublic?: boolean`, meaning the property may or may not be present, but if present is a boolean
* vs. `isWebPublic: boolean | void`, meaning the property is always present, but may be `undefined` as well as a boolean

I think the right answer in each of these cases is `boolean | void`, because in fact we're always passing the prop (or in the case of constructing an `UnreadStreamItem`, always setting the property), just with a value that may be `undefined` because it comes from getting a property (off of a Stream or Subscription object) that may not exist.

---

That made me curious whether the neighboring props here like `isSubscribed` should also have `| void` rather than `?:`. But in fact, if you do a search like `git grep -B2 -A18 '<StreamItem'` to find all the call sites of this component, you find that we sometimes omit those props entirely, so for them `?:` is accurate.

(I think the fact we sometimes leave those out is at least in part a symptom of some misfactoring -- #3767 -- but that's another matter. I feel like I've bumped into that particular misfactoring a few different times in the past month, so it might be something I pick up and deal with soon.)

Reply:
I changed them all to `boolean | void`. There's a small part of me that thinks it could be onerous to plumb all values as either value or undefined all the way down to all the code that changes and new data that is added to the api from now until the end of time. But I suppose if we stay on top of updating the code once old versions fall out of support, it's not so bad.
It's definitely nice that it's explicit and not just defaulted somewhere closer to the API so that the code that actually reads it can make better decisions.

Reply:
Yeah. Also I think the number of places you had to plumb this new field through is, fortunately, atypical for our codebase; see comment below at https://github.com/zulip/zulip-mobile/pull/5238#discussion_r808412840 .